src/@3rdweb-sdk/react/hooks/useEngine.ts

```diff
@@ -1270,3 +1278,28 @@ export function useEngineSubscriptionsLastBlock(
     },
   );
 }
+
+export interface EngineResourceMetrics {
+  error: string;
+  data: {
+    cpu: number;
+    memory: number;
+  };
+}
+
+export function useEngineResourceMetrics(engineId: string) {
+  return useQuery(engineKeys.metrics(engineId), async () => {
```

Comment:
this will query it once, I would think it would be more useful to poll it maybe?

You could do that by adding a `refetchInterval` prop on the query

Reply:
Good call, it's currently a single fetch